huaweicloud/services/apig/resource_huaweicloud_apig_custom_authorizer.go

```diff
@@ -0,0 +1,264 @@
+package apig
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/huaweicloud/golangsdk/openstack/apigw/v2/authorizers"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+)
+
+const (
+	FRONT_AUTH   = "FRONTEND"
+	BACKEND_AUTH = "BACKEND"
+)
+
+func ResourceApigCustomAuthorizerV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceApigCustomAuthorizerV2Create,
+		Read:   resourceApigCustomAuthorizerV2Read,
+		Delete: resourceApigCustomAuthorizerV2Delete,
+		Importer: &schema.ResourceImporter{
+			State: resourceApigCustomAuthorizerResourceImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[A-Za-z][\\w0-9]{2,63}$"),
```

Comment:
Suggest to use `^[A-Za-z][\\w]{2,63}$` for more conciseness.

Reply:
thx, the `\w` contain `0-9`.